ferrocene/doc/self-test-requirements/src/requirements.rst

```diff
@@ -0,0 +1,154 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Ferrocene Developers
+
+.. default-domain:: qualification
+
+General checks
+==============
+
+Checks for executables
+----------------------
+
+.. note::
+
+   **executable** refers to the following:
+
+   - ``rustc``
+   - ``rustdoc``
+
+The executable should exist in the ``bin`` directory, relative to the root of the installation.
+
+The metadata of the executable should be fetchable.
+
+The executable should be a regular file.
+
+The executable should have read and execute permissions set for all users.
+
+When executed with ``-vV`` command-line options,
+the executable should exit successfully.
+
+When executed with ``-vV`` command-line options,
+the output of the executable should be UTF-8 text.
+
+When executed with ``-vV`` command-line options,
+the output of the executable should have at least 3 values: host, commit-hash, and release.
+
+When executed with ``-vV`` command-line options,
+the value of ``host`` from the output of the executable should match the target-triple
+of the executing environment.
+
+When executed with ``-vV`` command-line options,
+the value of ``commit-hash`` from the output of the executable should match the 
+commit hash of the Ferrocene release corresponding to the installed Ferrocene toolchain.
+
+When executed with ``-vV`` command-line options,
+the value of ``release`` from the output of the executable should match the
+version of the Ferrocene release corresponding to the installed Ferrocene toolchain.
```

Comment:
Probably here we can use subrequirements as well. 

```suggestion
When executed with ``-vV`` command-line options:

* the executable should exit successfully.
* the output of the executable should be UTF-8 text.
* the output of the executable should have at least 3 values: host, commit-hash, and
  release.
* the value of ``host`` from the output of the executable should match the target-triple
  of the executing environment.
* the value of ``commit-hash`` from the output of the executable should match the 
  commit hash of the Ferrocene release corresponding to the installed Ferrocene toolchain.
* the value of ``release`` from the output of the executable should match the
  version of the Ferrocene release corresponding to the installed Ferrocene toolchain.
```

Reply:
c935a1ec0b2